python/src/trezorlib/stellar.py

```diff
@@ -97,13 +98,25 @@ def parse_transaction(xdr: str, network_passphrase: str = DEFAULT_NETWORK_PASSPH
 
     memo = parsed_tx.memo
     if isinstance(memo, TextMemo):
-        tx.memo_text = memo.memo_text
-    if isinstance(memo, IdMemo):
+        # In the ledger, memo_text is stored in bytes, so in theory, memo_text should be defined as bytes.
+        # For forward compatibility, we try to decode here. If the decoding fails, a decoding error will be
+        # displayed on the device.
+        try:
+            tx.memo_type = MEMO_TYPE_TEXT
+            tx.memo_text = memo.memo_text.decode("utf-8")
+        except UnicodeDecodeError:
+            tx.memo_text = "Unable to display the text memo"
```

Comment:
This seems trivially incorrect - you're changing the transaction context. Trezor will happily sign a transaction with the memo `"Unable to display the text memo"`, but it's probably a different transaction from the one submitted.

Reply:
I haven't read the firmware code yet, do you mean that Trezor will return the wrong transaction signature in this case?

Reply:
It's the right transaction signature for what you're submitting ;) but it's wrong for the decoded transaction.

Also the Laboratory claims that "Unable to display the text memo" is too long. I'm wondering where this limit is enforced, and whether we should also validate it on the firmware side.

Reply:
I checked the code and it seems to have an impact, so in this case, can we throw an exception first and prompt that the transaction cannot be processed? Then we fix it in another PR.

Reply:
this whole snippet, including the comment, is very confusing and seems wrong.

In XDR the memo text is defined as `string`, which per XDR RFC must be ASCII, per Stellar docs probably UTF. In any case what you get from the parsed XDR should be of type `str`, right? If the XDR parser returns `bytes`, then the `decode()` call is correct, but ... if the decoding fails we need to raise an exception anyway because the incoming XDR is incorrect. So maybe remove the whole try/except block and let the original UnicodeDecodeError fall through?

also the "forward-compatibility" is probably the wrong term here. Suggest only something like:
```
# memo_text is specified as UTF-8 string, but returned as bytes from the XDR parser
```

Reply:
> Then we fix it in another PR (fix memo_text to bytes type.).

`memo_text` is [defined as `string`](https://github.com/stellar/stellar-core/blob/master/src/xdr/Stellar-transaction.x#L520) so I believe there's nothing to fix on the firmware side

Reply:
Thanks for your explanation, I don't want this to cause the entire PR to be blocked, so I removed it first. I will make an in-depth test later.

4ad2a3e496abd530dcc142d0614d6630cdf15f48